### ISSUE TITLE
tags (taxonomy, not HTML) not saved with post

### ISSUE THREAD

**Issue by pbiron** [WordPress/gutenberg#7735]
I'm not sure if this is related to #5879 or not, but when I was preparing https://make.wordpress.org/core/2018/07/04/4-9-8-schedule/ yesterday the tags I'd assigned to the post were showing up in the editor but did not display when viewing the published post.

**To Reproduce**
1. Login in to make/core as user with `author` role
2. Create/edit a post
3. Assign tags (in my case, both where pre-existing tags)
4. Save post as draft
5. See that assigned tags are are listed in the `Tags` panel
6. Publish the post
7. View the published post and see that the assigned tags do not display

**Expected behavior**
I expected the assigned tags to display in the published post.

**Work Around**
As a work around (suggested to me by @SergeyBiryukov), I switched to the class editor, re-assigned the tags and updated the post...and then the assigned tags display in the published post.

**Comment by designsimply** [comment]
I tested with WordPress 4.9.8 and Gutenberg 3.5.0 using Firefox 61.0.2 on macOS 10.13.6 while logged in as the author role using the steps provided and was unable to reproduce the problem you described where tags were visible in a draft but gone after publishing. ([1m12s](https://cloudup.com/cTXijXWmrlm))

I also tried creating a draft post on make.wordpress.org/core using the administrator role and saving tags then closing and re-opening (but did not publish), and couldn't trigger the problem in a draft—not a perfect test though because I didn't do the publish step:

1. Go to Posts > Add New.
1. Add a title, content, and some pre-existing tags.
1. Click the "Save Draft" button.
1. Check to see if the assigned tags appear in the Document Settings > Tags panel.
1. Close the post and go to Posts.
1. Check to see if the assigned tags appear in the posts list.
1. Re-open the post for editing.
1. Check to see if the assigned tags still appear in the Document Settings > Tags panel.

**Result:** tags appeared normally in the draft, in the posts list, and after re-opening the draft. ([53s](https://cloudup.com/cmuXQiDhRNl))

@pbiron would it be possible for you to test this again and note the WordPress, Gutenberg, and browser versions you are using at the time of your test? Do you happen to know if there any plugins are installed on make.wordpress.org/core which could be relevant?

Testing note: would be good to loop back around and test with master, with multisite, and using different roles.

**Comment by pbiron** [comment]
At the time I created this issue I was using Chrome 67 on Win10 Pro (Chrome 68 now).  Can't be sure, but given the timing, I think make/core was using Gutenberg 3.1 (or some _vNext_ nightly build) at the time.  As far as I know, make/core uses the "bleeding edge" nightly build of WP at any given time.  I have no way of knowing what other plugins were installed on make/core at the time.

Also, since I first experienced this problem on make/core my account has gone from Author -> Editor so if the author role is a key part of the problem I can't test that any more (at least on make/core).  I edited your draft post on make/core and was able to successfully add tags and re-save as draft.

As mentioned in the original description of this issue, at the time @SergeyBiryukov told me
> Yeah, there's a Gutenberg issue with saving tags

@audrasjb also told me:
> I already had an issue with tags

I will say that since opening this issue, I have installed various combinations of GB 3.3, 3.4 and 3.5 on WP 4.9.7 and 4.9.8 (including each of the betas/RCs) on several sites I use for testing and haven't noticed the problem on those sites.

So, maybe the underlying problem got fixed (without trying?) in a subsequent GB release.

I'll be doing another make/core post in a week or so and will report back here as to how that goes.
